File: 17-Chain-of-Responsibility/src/go/manager.go
package main

import "fmt"

type Manager struct {
	BaseApprover
}

func (m *Manager) Handle(request PurchaseRequest) {
	if request.Amount <= 5000 {
		fmt.Printf("  Manager approved $%.2f for %s\n", request.Amount, request.Purpose)
	} else {
		m.passToNext(request)
	}
}

File: 17-Chain-of-Responsibility/src/java/chapter17/Manager.java
package chapter17;

public class Manager extends Approver {
    public Manager() {
        super("Manager", 5000);
    }

    @Override
    protected void approve(PurchaseRequest request) {
        System.out.printf("  %s approved $%.2f for %s%n",
                name, request.getAmount(), request.getPurpose());
    }
}
